src/internal/components/dropdown/index.tsx

```diff
@@ -197,6 +197,14 @@ const Dropdown = ({
     } else {
       target.style.width = position.width;
     }
+
+    // Prevent the dropdown width from stretching beyond the trigger width
+    // if that is going to cause the dropdown to be cropped because of overflow
+    if (position.overflows) {
+      target.classList.remove(styles['stretch-beyond-trigger-width']);
+      target.style.removeProperty('maxWidth');
```

Comment:
Why maxWidth is removed?

Reply:
Because when `stretchBeyondTriggerWidth` is `true`, then both the `stretch-beyong-trigger-width` class and the `maxWidth` are applied (see line 106 [here](https://github.com/cloudscape-design/components/blob/main/src/internal/components/dropdown/index.tsx#L106)) and line 116 [here](https://github.com/cloudscape-design/components/blob/main/src/internal/components/dropdown/index.tsx#L118) respectively). Here we need to revert both changes.

Why is `maxWidth` applied separately from the class in the first place? I assume basically because its value comes from JavaScript, [here](https://github.com/cloudscape-design/components/blob/main/src/internal/breakpoints.ts#L11). I guess this could be refactored, e.g, by setting a CSS custom property that could be used in a ruleset under the class selector. This way adding and removing the class would be enough, and this part in particular might be a bit more readable but at the same time this would add a bit more complexity overall, for just removing one line, so not sure it's worth it.

Reply:
I think eventually we need to refactor that entire component but that's not in scope. What I would recommend is to add a code comment to explain why the max-width is removed.